library/core/src/primitive_docs.rs

```diff
@@ -1527,6 +1527,76 @@ mod prim_ref {}
 /// Note that all of this is not portable to platforms where function pointers and data pointers
 /// have different sizes.
 ///
+/// ### ABI compatibility
+///
+/// Generally, when a function is declared with one signature and called via a function pointer with
```

Comment:
I feel like this section is missing a prologue that says that function signature casting is wildly unsafe and you should use matching function pointers in almost all situations rather than tempt fate and UB in this way.

Reply:
Good point, I added such a paragraph. Suggestions for better wording are always welcome!